st2actions/st2actions/runners/mistral/v2.py

```diff
@@ -144,7 +144,10 @@ def try_run(self, action_parameters):
         inputs = self.runner_parameters.get('context', dict())
         inputs.update(action_parameters)
 
-        endpoint = 'http://%s:%s/v1/actionexecutions' % (cfg.CONF.api.host, cfg.CONF.api.port)
+        # If host value does not start with http, then default to http://.
+        protocol = 'http' if not cfg.CONF.api.use_ssl else 'https'
```

Comment:
To clarify it - if we update this place to use `get_full_public_api_url` we can get rid of this new (and potentially confusing) option all together.

We just need to make sure `cfg.CONF.auth.api_url` option is configured correctly, but this should already be the case otherwise other things wouldn't work.